packages/common/src/store/pages/chat/middleware.ts

```diff
@@ -42,13 +43,20 @@ export const chatMiddleware =
             )
           }
           reactionListener = ({ chatId, messageId, reaction }) => {
-            store.dispatch(
-              setMessageReactionSucceeded({
-                chatId,
-                messageId,
-                reaction
-              })
+            const message = getChatMessageById(
```

Comment:
I don't know why, but it feels strange to put this logic in the middleware. Could it not live in the slice?
Seems like it's not for _writing_ a reaction, but for adding a reaction that we received over the socket to a chat that we have locally?

Reply:
feel similarly, could we not just return early if `existingMessage` doesn't exist here?
https://github.com/AudiusProject/audius-client/blob/e25bb63df325e1b21249c249ec9ce7ffd2d21baa/packages/common/src/store/pages/chat/slice.ts#L281

Reply:
Yea y'all are right, will change!